### PR TITLE
Migration of test_xml_dump_of_gluster_volume_status_during_rebalance

### PR DESCRIPTION
**Pull request by nishith-vihar**
[test case](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py) is migrated from glusto to redant and get_volume_status is modified in volume_ops to accomodate the tasks option in status

Updates: #292
Fixes: #536
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

### CODE REVIEW

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                                                  self.server_list[0])
+        status = status_info['aggregate']['statusStr']
+
+        if 'in progress' not in status:
+            raise Exception("Rebalance process is not running")
+
+        redant.logger.info("Rebalance process is running")
+
+        # Stop glusterd
+        redant.stop_glusterd(self.server_list[2])
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:103]: check if glusterd is stopped here. It'll take time for a big daemon like glusterd to go down and also take its processes down.

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 1 above)
Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:103]: that time might not be big, but wrt test case logic is some validation depends on glusterd being down and glusterd is still up then things will fail.

Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:103]: Ack

File: common/ops/gluster_ops/volume_ops.py [hunk 2]
```diff
@@ -867,9 +880,9 @@ def wait_for_volume_process_to_be_online(self, volname: str, node: str,
             return False
 
         # Wait for bricks to be online
-        bricks_online_status = (self.wait_for_bricks_to_come_online(volname,
-                                server_list,
-                                brick_list))
+        bricks_online_status = (
+            self.wait_for_bricks_to_come_online(volname, server_list,
```
Comment (nik-redhat) [inline common/ops/gluster_ops/volume_ops.py:884]: Hey why this indentation change?

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+from tests.d_parent_test import DParentTest
+
+# disruptive;dist-rep
+
+
+class TestCase(DParentTest):
+    """
+    xml Dump of gluster volume status during rebalance, when one gluster
+    node is down
+    """
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:33]: nit: can be removed

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                                                      counter, 2, 3, 4, 10,
+                                                      mount['client'])
+            self.all_mounts_procs.append(proc)
+            counter = counter + 10
+
+        # Validate IO
+        if not redant.validate_io_procs(self.all_mounts_procs, self.mounts):
+            raise Exception("IO failed on some of the clients")
+
+        ret = redant.volume_stop(self.volume_name1, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:87]: nit: ret not needed

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Start Rebalance
+        ret = redant.rebalance_start(self.vol_name, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:90]: Same as above

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Get rebalance status
+        status_info = redant.get_rebalance_status(self.vol_name,
+                                                  self.server_list[0])
+        status = status_info['aggregate']['statusStr']
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:95]: Better to check for `None` then derefence

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        if not redant.wait_for_glusterd_to_stop(self.server_list[2]):
+            raise Exception("glusterd has not stopped on "
+                            f"{self.server_list[2]}")
+
+        cmd = ("gluster v status  | grep -A 4 'Rebalance' | awk 'NR==3{print "
+               "$3,$4}'")
+        ret = redant.execute_abstract_op_node(cmd, self.server_list[0])
+        ret1 = redant.get_volume_status(self.vol_name, self.server_list[0],
+                                        options="tasks")
+        rebalance_status = ret1[self.vol_name]['task_status'][0]['statusStr']
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:113]: Again check for `None` will be helpful.

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Stop glusterd
+        redant.stop_glusterd(self.server_list[2])
+        if not redant.wait_for_glusterd_to_stop(self.server_list[2]):
+            raise Exception("glusterd has not stopped on "
+                            f"{self.server_list[2]}")
+
+        cmd = ("gluster v status  | grep -A 4 'Rebalance' | awk 'NR==3{print "
+               "$3,$4}'")
+        ret = redant.execute_abstract_op_node(cmd, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:110]: Check the `error_code` of the command execution after this.

**Review by nik-redhat — CHANGES_REQUESTED** [review]

File: common/ops/gluster_ops/volume_ops.py (hunk 2 above)
Comment (nishith-vihar) [reply common/ops/gluster_ops/volume_ops.py:884]: It is giving flake error

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 3 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:33]: Ack

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 4 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:87]: Ack

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 5 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:90]: Ack

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 6 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:95]: Ack

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 7 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:113]: Ack

File: tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py (hunk 8 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_xml_dump_of_gluster_volume_status_during_rebalance.py:110]: error_code will be checked in the abstract_ops only

**Comment by nishith-vihar** [comment]
```
test_xml_dump_of_gluster_volume_status_during_rebalance-dist-rep PASS

Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_xml_dump_of_gluster_volume_status_during_rebalance
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|   dist-rep  |     PASS    | 132.73854088783264 |
+-------------+-------------+--------------------+



Framework runtime : 136.58933687210083
```

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**